### ISSUE TITLE
Absence of g++ results in "successful" build

### ISSUE THREAD

**Issue by afaerber** [ARM-software/ComputeLibrary#18]
If `sconscript` catches an exception trying to execute `g++`, it prints an error but does not `Exit(1)` as done elsewhere. That leads to `scons` returning success (0) despite not actually building anything, which is confusing in an unattended package build scenario such as OBS.

```
try:
    compiler_ver = subprocess.check_output( [env['CXX'] , "-dumpversion"] ).strip()
except OSError:
    print "ERROR: Compiler not found"
    compiler_ver = ""

if compiler_ver != "":
    ...
```

**Comment by AnthonyARM** [comment]
Good point, will fix this in the next release

**Comment by AnthonyARM** [comment]
Fixed in v17.04 (Re-open if you still see the issue)
